### DISCUSSION TITLE
Support for Ferris Swoop keyboard

### DISCUSSION THREAD

**Discussion by adnanaq** [manna-harbour/miryoku#356] [Q&A] [answered]
Hi,

This is request to add support for ferris [swoop](https://github.com/jimmerricks/swoop) keybaord. I have a working firmware for it https://github.com/adnanaq/swoop-zmk-config. It's fork of https://github.com/snicklepickles/swoop-zmk-config, which was outdated.

**Comment by manna-harbour** [comment, marked as answer]
Here's the change in a temporary branch: https://github.com/manna-harbour/miryoku_zmk/commit/c159cc94fce877bbb4ac97e6eb49491ed7e941fd
Here's a sample build run: https://github.com/manna-harbour/miryoku_zmk/actions/runs/11657488218
Verify that it works and I'll add it to the main branch.

**Reply by pestoAioli** [reply]
hey any updates on whether this is gonna get added to the main branch? it worked great for me!

**Reply by adnanaq** [reply]
Yeah, I am able to get it to work on my end too. It wasn't merged yet because I was testing few other things. But it is good to go. @manna-harbour merge ir whenever you can.

**Reply by manna-harbour** [reply]
Merged now.

**Reply by alex-grover** [reply]
should the swoop be added to https://github.com/manna-harbour/miryoku/discussions/81?

**Reply by manna-harbour** [reply]
@alex-grover Added now.

**Comment by adnanaq** [comment]
I had to build the firmware with windows config, and it appears everything is working except the RGB control. Nothing happens when I set RGB on/off or any correspondent settings.

EDIT: Had to add related settings in config. But now I [see](https://github.com/adnanaq/miryoku_zmk/actions/runs/11707786942/job/32607980534#step:6:282) `#error "A zmk,underglow chosen node must be declared"`

**Reply by manna-harbour** [reply]
Yes you have to build with the appropriate Kconfig to enable RGB. The shield definition also needs to support RGB, so if there's a build error it might need updating.

**Reply by adnanaq** [reply]
Can you point me what I need to define  in my workflow input to make RGB work?

**Reply by manna-harbour** [reply]
For the workflow, you need e.g. `CONFIG_ZMK_RGB_UNDERGLOW=y\nCONFIG_ZMK_RGB_UNDERGLOW=y` in the kconfig field. See https://zmk.dev/docs/features/underglow and https://github.com/manna-harbour/miryoku_zmk#kconfig.

**Reply by adnanaq** [reply]
Yeah. I added those configs during workflow build. And it's same [error]. `#error "A zmk,underglow chosen node must be declared"`(https://github.com/adnanaq/miryoku_zmk/actions/runs/11940863849/job/33284472401#step:6:1916).

**Reply by manna-harbour** [reply]
The shield definition also needs to support RGB, so if there's a build error it might need updating.

See https://zmk.dev/docs/features/underglow#adding-rgb-underglow-to-a-board.

**Reply by adnanaq** [reply]
So, if I understand correctly, I need to add underglow/rgb config in your cloned repo to make it work?

Edit: I am able to get it to work. You can merge it.
